### PR TITLE
Modeler 2.5a: Read models and system yaml files [ANT-2306]

### PR DESCRIPTION
**Pull request by payetvin**
Add the main modeler as a new binary
Directory loadFiles added, used to call previous work to read yaml files by handling I/O and errors related to yaml parsing

### CODE REVIEW

File: src/solver/modeler/loadFiles/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+
+# Link dependencies (if any)
+target_link_libraries(loadModelerFiles
+        PUBLIC
+        Antares::antares-study-system-model
+        PRIVATE
+        io
```
Comment (flomnes) [inline src/solver/modeler/loadFiles/CMakeLists.txt]:
```suggestion
        Antares::io
```

File: src/solver/modeler/loadFiles/readLibraries.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+std::vector<Study::SystemModel::Library> loadLibraries(const fs::path& studyPath)
+{
+    std::vector<Study::SystemModel::Library> libraries;
+
+    const fs::path directoryPath = studyPath / "input" / "model-libraries";
+    for (const auto& entry: fs::directory_iterator(directoryPath))
+    {
+        if (entry.path().extension() != "yml")
+        {
+            logs.info() << "File ignored because of wrong extension: " << entry.path();
```
Comment (flomnes) [inline src/solver/modeler/loadFiles/readLibraries.cpp]: OK but what is a valid extension ?
```suggestion
            logs.info() << entry.path() " ignored, only files having the `yml` extension are loaded";
```

**Review by flomnes — COMMENTED** [review]
Good start ! Now we need a `main`

File: src/solver/modeler/loadFiles/CMakeLists.txt (hunk 1 above)
Comment (payetvin) [reply src/solver/modeler/loadFiles/CMakeLists.txt]: doesn't exists

Comment (payetvin) [reply src/solver/modeler/loadFiles/CMakeLists.txt]: Added this in io/CmakeLists.txt `add_library(Antares::io ALIAS io)`

File: src/solver/modeler/loadFiles/readParameters.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ModelerParameters loadParameters(const fs::path& studyPath)
+{
+    try
+    {
+        const std::string paramStr = IO::readFile(studyPath / "parameters.yml");
+        return parseModelerParameters(paramStr);
+    }
+    catch (const YAML::Exception& e)
+    {
+        logs.error() << "Error while parsing the yaml parameters file";
```
Comment (flomnes) [inline src/solver/modeler/loadFiles/readParameters.cpp]:
```suggestion
        const auto fullPath = studyPath / "parameters.yml";
        const std::string paramStr = IO::readFile(fullPath);
        // ...
        logs.error() << "Error while parsing parameter file " << fullPath;
```

**Review by flomnes — APPROVED** [review]

**Pull request merged**